pandas/_libs/window/aggregations.pyx

```diff
@@ -1752,6 +1752,51 @@ def roll_weighted_var(float64_t[:] values, float64_t[:] weights,
 # ----------------------------------------------------------------------
 # Exponentially weighted moving average
 
+def ewma_time(ndarray[float64_t] vals, int minp, ndarray[int64_t] times,
+              int64_t halflife):
+    """
+    Compute exponentially-weighted moving average using halflife and time
+    distances.
+
+    Parameters
+    ----------
+    vals : ndarray[float_64]
+    minp : int
+    times : ndarray[int64]
+    halflife : int64
+
+    Returns
+    -------
+    ndarray
+    """
+    cdef:
+        Py_ssize_t i, num_not_nan = 0, N = len(vals)
+        bint is_not_nan
+        float64_t last_result
+        ndarray[uint8_t] mask = np.zeros(N, dtype=np.uint8)
+        ndarray[float64_t] weights, observations, output = np.empty(N, dtype=np.float64)
+
+    if N == 0:
+        return output
+
+    last_result = vals[0]
+
+    for i in range(N):
```

Comment:
can you you put this in a nogil block? do we have timings on this?

Reply:
I tried to `nogil` this but could not since we are doing array arithmetic & broadcasting.

```
            weights = 0.5 ** ((times[i] - times[mask.view(np.bool_)]) / halflife)
            observations = vals[mask.view(np.bool_)]
            last_result = np.sum(weights * observations) / np.sum(weights)
```

I can add an ASV for this.

Reply:
ok sure on the asv.

I think you can convert the mask view to a np.take operation (but can always just add an issue for this) if needed